runtime/src/snapshot_utils.rs

```diff
@@ -476,6 +603,20 @@ where
     )
 }
 
+fn deserialize_snapshot_data_files<F, T>(
+    snapshot_root_paths: &SnapshotRootPaths,
+    deserializer: F,
```

Comment:
nit: throughout the codebase we've been using `deserializer: impl FnOnce...` instead of the `where` clause below. Might be good to keep that consistent across all the functions here.

Reply:
Ah OK. I was following the existing impl of the file; I'll happily make these change though!

Reply:
Done.